code/datums/components/magic_item.dm

```diff
@@ -0,0 +1,33 @@
+//The base component for magical items.
+//Allows for item agnostic handling, and avoids needing to change the base item code
+//Most of our logic should be contained in the actual effects
+
+/datum/component/magic_item
```

Comment:
I would suggest a signal for adding an effect post creation so silver weapons could say be enchanted,

Reply:
also there already is an enchantment system in the game that supports weighted enchantments and stuff from the tarkov inventory

Reply:
The inventory PR thats getting reverted?

Reply:
I assumed it was still in because there are lingering files left over. But yea original point stands it should have a signal to add new enchantments. The alternative of reading the component is weird, I don't even really know why its a component to begin with tbh